test/serializers/key_format_test.rb

```diff
@@ -0,0 +1,48 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class KeyFormatTest < Minitest::Test
+      def setup
+        @author = Author.new(name: 'Steve K.')
+
+        @first_post = Post.new(
+                        id:                   1,
+                        title:                'Hello!!',
+                        body:                 'Hello, world!!',
+                        multi_word_attribute: 'Something about hello')
+        @second_post = Post.new(
+                        id:                   2,
+                        title:                'New Post',
+                        body:                 'Body',
+                        multi_word_attribute: 'Something about new')
+
+        @author.posts = [@first_post, @second_post]
+
+        @first_post.author = @author
+        @second_post.author = @author
+
+        @first_post.comments = []
+        @second_post.comments = []
+      end
+
+      def test_overrides_default_adapter_key_format
+        @serializer = PostWithKeyFormatSerializer.new(@first_post)
+        @adapter = ActiveModel::Serializer::Adapter::JsonApi.new(@serializer)
```

Comment:
For future-proofing and consistency, we may want to instead do it the way the controller does (IIRC)

```ruby
serializable_resource = ActiveModel::SerializableResource.serialize(@first_post, adapter: :json_api, serializer: PostWithKeyFormatSerializer)
serializable_resource.as_json
```

e.g. see https://github.com/rails-api/active_model_serializers/blob/b78126565c7c8d29102cc8a8adf66ddf228ff032/test/serializers/meta_test.rb#L115-L117

```ruby
      def load_adapter(options)
        options = options.merge(adapter: :json-api, serializer: PostWithKeyFormatSerializer)
        ActiveModel::SerializableResource.new(@first_post, options)
      end
```